rgw/admin/bucket.go

```diff
@@ -132,7 +134,7 @@ func (api *API) GetBucketPolicy(ctx context.Context, bucket Bucket) (Policy, err
 
 // RemoveBucket will remove a given token from the object store
 func (api *API) RemoveBucket(ctx context.Context, bucket Bucket) error {
-	_, err := api.call(ctx, http.MethodDelete, "/bucket", valueToURLParams(bucket))
+	_, err := api.call(ctx, http.MethodDelete, "/bucket", valueToURLParams(bucket, []string{"bucket", "purge-objects"}))
```

Comment:
Is this purging all the objects from the bucket? If so, isn't that dangerous? Would it be good to have validation or something passed to the API by the user to do this?

Reply:
I did not touch change the behaviour of this API call. I only added the filtering in `valueToURLParams` (so only those parameters that are actually supported by the API according to the ceph-rgw source code are serialized to the query string). 

It is certainly a dangerous operation (even without purge objects, given that all obejcts that are only in this bucket will be deleted independently of the `purge-objects` setting), but the way it behaves does not change with this commit. 

With `purge-objects` set to true all the objects in the bucket will be deleted, before deleting the bucket (I guess that means they will also be deleted from other buckets that may contain them).

For purge-objects to happen `*bucket.PurgeObjects` must be set to true, this will never be the case for Bucket objects returned from the API, so explicit action is necessary on the side of the user.